src/pydata_sphinx_theme/assets/scripts/pydata-sphinx-theme.js

```diff
@@ -665,84 +668,76 @@ async function fetchAndUseVersions() {
 }
 
 /*******************************************************************************
- * Add keyboard functionality to mobile sidebars.
- *
- * Wire up the hamburger-style buttons using the click event which (on buttons)
- * handles both mouse clicks and the space and enter keys.
+ * Sidebar modals (for mobile / narrow screens)
  */
 function setupMobileSidebarKeyboardHandlers() {
-  // These are hidden checkboxes at the top of the page whose :checked property
-  // allows the mobile sidebars to be hidden or revealed via CSS.
-  const primaryToggle = document.getElementById("pst-primary-sidebar-checkbox");
-  const secondaryToggle = document.getElementById(
-    "pst-secondary-sidebar-checkbox",
+  // These are the left and right sidebars for wider screens. We cut and paste
+  // the content from these widescreen sidebars into the mobile dialogs, when
+  // the user clicks the hamburger icon button
+  const primarySidebar = document.getElementById("pst-primary-sidebar");
+  const secondarySidebar = document.getElementById("pst-secondary-sidebar");
+
+  // These are the corresponding left/right <dialog> elements, which are empty
+  // until the user clicks the hamburger icon
+  const primaryDialog = document.getElementById("pst-primary-sidebar-modal");
+  const secondaryDialog = document.getElementById(
+    "pst-secondary-sidebar-modal",
   );
-  const primarySidebar = document.querySelector(".bd-sidebar-primary");
-  const secondarySidebar = document.querySelector(".bd-sidebar-secondary");
-
-  // Toggle buttons -
-  //
-  // These are the hamburger-style buttons in the header nav bar. When the user
-  // clicks, the button transmits the click to the hidden checkboxes used by the
-  // CSS to control whether the sidebar is open or closed.
-  const primaryClickTransmitter = document.querySelector(".primary-toggle");
-  const secondaryClickTransmitter = document.querySelector(".secondary-toggle");
+
+  // These are the hamburger-style buttons in the header nav bar. They only
+  // appear at narrow screen width.
+  const primaryToggle = document.querySelector(".primary-toggle");
+  const secondaryToggle = document.querySelector(".secondary-toggle");
+
+  // Cut nodes and classes from `from`, paste into/onto `to`
+  const cutAndPasteNodesAndClasses = (from, to) => {
+    Array.from(from.childNodes).forEach((node) => to.appendChild(node));
+    Array.from(from.classList).forEach((cls) => {
+      from.classList.remove(cls);
+      to.classList.add(cls);
+    });
+  };
```

Comment:
Do you want to move that to a top level function ? I don't know the best JS practices, but I don't think we close over any variables. 

If you prefer to keep it inside setupMobileSidebarKeyboardHandlers that's fine too.